telos-frontend/src/views/journal/Journal.jsx

```diff
@@ -48,17 +51,28 @@ const Journal = () => {
     setDateLeftPage(selectedDate.getTime() - DAYS_TO_CHANGE_BY * MILLISECONDS_PER_DAY);
   }
 
+  // Formats date in milliseconds to YYYY-MM-DD format.
   function formatDateString(dateMilliseconds) {
     const date = new Date(dateMilliseconds);
     return new Date(date.getTime() - date.getTimezoneOffset() * 60000).toISOString().split('T')[0];
   }
 
-  const addNewWidget = (widgetName) => {
-    if (isRight) {
-      setWidgetsRight([...widgetsRight, { widgetType: widgetName, top: 0, left: 0 }]);
-    } else {
-      setWidgetsLeft([...widgetsLeft, { widgetType: widgetName, top: 0, left: 0 }]);
-    }
+  const addNewWidget = (type) => {
+    const date = isRight ? formatDateString(dateRightPage) : formatDateString(dateLeftPage);
+
+    const payload = {
+      type,
+      date,
+      position: { row: 0, col: 0 },
+    };
+
+    axios.post('http://localhost:3001/api/journal', payload).then(({ data }) => {
```

Comment:
Can we be sure this will always be on localhost:3001. Should we defer binding and have the url of the back end in a file that can be read, or a global var.

Reply:
I have now added a proxy in package.json that points to the backend server.

Reply:
Excellent